### PR TITLE
Fix validationresponse ticket close

### PR DESCRIPTION
**Pull request by Lainow**
## Checklist before requesting a review

*Please delete options that are not relevant.*

- [x] I have read the CONTRIBUTING document.
- [x] I have performed a self-review of my code.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] This change requires a documentation update.

## Description

- It fixes !35135
- Here is a brief description of what this PR does

When the ticket is closed and there are validation requests waiting for a response, it is still possible to validate them, but this does not update the global status.

So I deactivated the possibility of accepting a validation request once the ticket has been closed.

## Screenshots (if appropriate):
**Before**
image

**After**
image

### CODE REVIEW

**Review by cedric-anne — COMMENTED** [review]
I am not sure we should accept answer to validation requests once a ticket is closed.

**Comment by Lainow** [comment]
> I am not sure we should accept answer to validation requests once a ticket is closed.

Yes, I forgot to mention this in the description, but I've deactivated it now.

**Review by Rom1-B — CHANGES_REQUESTED** [review]
Seems OK.

Perhaps we should also exclude tickets closed on the “Tickets waiting for your approval” counter and its associated search:

image

File: src/CommonITILValidation.php
Comment (Rom1-B) [inline src/CommonITILValidation.php:686]: I think the query is wrong (before your PR), because it counts the number of pending validations, when it should count the number of tickets with (one or more) pending validations.
Deleted tickets should also be excluded.

**Review by Rom1-B — CHANGES_REQUESTED** [review]

File: src/CommonITILValidation.php
Comment (Rom1-B) [inline src/CommonITILValidation.php:688]: It should do the same
```suggestion
                'NOT' => [
                    'status' => Ticket::getClosedStatusArray(),
                ],
```

**Review by Rom1-B — CHANGES_REQUESTED** [review]

**Review by Rom1-B — APPROVED** [review]

File: src/CommonITILValidation.php
Comment (AdrienClairembault) [inline src/CommonITILValidation.php:684]: Validation is also available on changes.

**Review by AdrienClairembault — CHANGES_REQUESTED** [review]
Seems OK, you just need to take `Change` into account.
Also, the new behaviors could be tested.

**Review by AdrienClairembault — APPROVED** [review]

File: front/ticket.form.php
Comment (cedric-anne) [inline front/ticket.form.php:90]:
```suggestion
```

File: phpunit/functional/CommonITILValidationTest.php
Comment (cedric-anne) [inline phpunit/functional/CommonITILValidationTest.php:476]:
```suggestion
```

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**
